app/repositories/VatSchemeRepository.scala

```diff
@@ -236,9 +239,45 @@ class VatSchemeRepository @Inject() (
       .toFuture()
       .map(_ => status)
 
-  private[repositories] def registrationSelector(regId: String, internalId: Option[String] = None) =
+  private[repositories] def registrationSelector(regId: String, internalId: Option[String] = None) = {
+
+    val baseFilter: Bson = registrationIdFilter(regId)
+
+    val regFilter: Bson = internalId match {
+      case Some(id) =>
+        and(
+          baseFilter,
+          internalIdFilter(id),
+          draftDateFilter
+        )
+      case None =>
+        baseFilter
+    }
+    regFilter
+  }
+
+  private[repositories] def registrationSelectorNewApp(regId: String, internalId: Option[String] = None) = {
     internalId
       .map(intId => and(equal("registrationId", regId), equal("internalId", intId)))
       .getOrElse(equal("registrationId", regId))
+  }
+
+  // Extract the document for a specific user(internalId) provided if its in submitted status or
+  // (any other status & application created on or after the cutOffDate)
+  private def internalIdAndDateFilterSelector(internalId: String) = {
+    val baseFilter: Bson = internalIdFilter(internalId)
+
+    val regFilter: Bson =
+      and (
+        baseFilter,
+        draftDateFilter
+      )
+    regFilter
+  }
+
+  private def internalIdFilter(internalId: String): Bson = equal(internalIdKey, internalId)
+  private def registrationIdFilter(registrationId: String): Bson = equal(registrationIdKey, registrationId)
+  private def draftDateFilter: Bson = or(and(exists("status", true), equal("status", "submitted")),
```

Comment:
If `equal("status", "submitted")` is true then `exists("status", true)` will have to be true. So having this as part of an `and(...)` seems redundant.
Is this what fixes the tests? In which case it feels a little dubious.
Do the tests work if you have:
`or(`
`equal("status", "submitted")`
`and(exists("status", true), notEqual ("status", "submitted"), Filters.gte ("createdDate", cutoffDate)))`
`)`